src/components/ProjectDomain/projectdomainTable.tsx

```diff
@@ -0,0 +1,302 @@
+import React, { useState, useEffect } from "react";
+import { PackageOpen, Eye, Trash2 } from "lucide-react";
+
+import { useToast } from "@/components/ui/use-toast";
+import AddProjectDomain from "@/components/ProjectDomain/addProjectDomain";
+import { Card } from "@/components/ui/card";
+import {
+  Table,
+  TableHeader,
+  TableBody,
+  TableRow,
+  TableHead,
+  TableCell,
+} from "@/components/ui/table";
+import {
+  Dialog,
+  DialogTrigger,
+  DialogContent,
+  DialogHeader,
+  DialogTitle,
+} from "@/components/ui/dialog";
+import {
+  Tooltip,
+  TooltipTrigger,
+  TooltipContent,
+} from "@/components/ui/tooltip";
+import { axiosInstance } from "@/lib/axiosinstance";
+import { Button } from "@/components/ui/button";
+import { Switch } from "@/components/ui/switch";
+import { statusType } from "@/utils/common/enum";
+import { apiHelperService } from "@/services/projectdomain";
+
+interface DomainData {
+  _id: string;
+  label: string;
+  description: string;
+  createdAt?: string; // ISO date string
+  createdBy?: string;
+  status?: string; // User or system that created the domain
+}
+
+const ProjectDomainTable: React.FC = () => {
+  const [domainData, setDomainData] = useState<DomainData[]>([]);
+  const [loading, setLoading] = useState(true);
+  const [noData, setNoData] = useState(false); // State to handle no data available
+  const { toast } = useToast();
+  // Function to fetch domain data
+  const fetchDomainData = async () => {
+    setLoading(true);
+    setNoData(false); // Reset noData state before fetching
+    try {
+      const response = await apiHelperService.getAllProjectdomain();
+      if (!response.data.data) {
+        setNoData(true); // Set noData if response is empty
+      } else {
+        setDomainData(response.data.data);
+      }
+    } catch (error) {
+      toast({
+        title: "Error",
+        description: "Failed to fetch domain data. Please Refresh the page.",
```

Comment:
I suppose this is a pretty common fallback case, create a message in enums with replacable resource type and use that string by replacing the resource name at runtime
Hope I am making sense, please connect with me if you are confused by what I am asking to do

Reply:
done with the changes created messages accordingly